core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7772CoreExtensionsNotFoundTest.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.maven.it;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.apache.maven.shared.verifier.VerificationException;
+import org.apache.maven.shared.verifier.Verifier;
+import org.apache.maven.shared.verifier.util.ResourceExtractor;
+import org.junit.jupiter.api.Test;
+
+/**
+ * This is a test set for <a href="https://issues.apache.org/jira/browse/MNG-7772">MNG-7772</a>:
+ * check that Maven fails if it cannot load core extensions contributed by <code>${user.home}/.m2/extensions.xml</code>.
+ */
+public class MavenITmng7772CoreExtensionsNotFoundTest extends AbstractMavenIntegrationTestCase {
+
+    public MavenITmng7772CoreExtensionsNotFoundTest() {
+        super("[3.9.0,)");
+    }
+
+    @Test
+    public void testCoreExtensionsNotFound() throws Exception {
+        File testDir = ResourceExtractor.simpleExtractResources(getClass(), "/mng-7772-core-extensions-not-found");
+
+        Path m2TestDirPath = Paths.get(testDir.toPath().toString(), ".m2");
+        Files.createDirectories(Paths.get(System.getProperty("user.home"), ".m2"));
+        Path to = Paths.get(System.getProperty("user.home"), ".m2", "extensions.xml");
+
+        // Test cannot run when there is already an extensions.xml file present in the ${user.home}/.m2 folder
+        assertFalse(Files.exists(to));
+
+        try {
+            Files.copy(m2TestDirPath.resolve("extensions.xml"), to);
+            Verifier verifier = newVerifier(testDir.getAbsolutePath());
+
+            try {
+                verifier.addCliArgument("validate");
+                verifier.execute();
```

Comment:
I would like to try:

```
        verifier.setSystemProperty("user.home", mocUserHom_with_m2 );

```

Reply:
yes I know it is not the best ... but better than interact with real user directory

Reply:
I see why it's not preferable to use the real user home. Somehow `.setSystemProperty("user.home"` was not working but I found that test MavenITmng0553SettingsAuthzEncryptionTest already had a method for setting the user home. I've reused that code and that's working :)